rust/parquet/src/file/writer.rs

```diff
@@ -521,6 +526,75 @@ impl<T: Write + Position> PageWriter for SerializedPageWriter<T> {
     }
 }
 
+struct ArrowWriter {
+    writer: SerializedFileWriter<File>,
+    rows: i64,
+}
+
+impl ArrowWriter {
+    pub fn new(file: File, _arrow_schema: &Schema) -> Self {
+        //TODO convert Arrow schema to Parquet schema
+        let schema = Rc::new(
+            types::Type::group_type_builder("schema")
+                .with_fields(&mut vec![
+                    Rc::new(
+                        types::Type::primitive_type_builder("a", Type::INT32)
+                            .with_repetition(Repetition::REQUIRED)
+                            .build()
+                            .unwrap(),
+                    ),
+                    Rc::new(
+                        types::Type::primitive_type_builder("b", Type::INT32)
+                            .with_repetition(Repetition::REQUIRED)
+                            .build()
+                            .unwrap(),
+                    ),
+                ])
+                .build()
+                .unwrap(),
+        );
+        let props = Rc::new(WriterProperties::builder().build());
+        let file_writer =
+            SerializedFileWriter::new(file.try_clone().unwrap(), schema, props).unwrap();
+
+        Self {
+            writer: file_writer,
+            rows: 0,
+        }
+    }
+
+    pub fn write(&mut self, batch: &RecordBatch) {
+        let mut row_group_writer = self.writer.next_row_group().unwrap();
+        for i in 0..batch.schema().fields().len() {
+            let col_writer = row_group_writer.next_column().unwrap();
+            if let Some(mut writer) = col_writer {
+                match writer {
+                    ColumnWriter::Int32ColumnWriter(ref mut typed) => {
+                        let array = batch
+                            .column(i)
+                            .as_any()
+                            .downcast_ref::<array::Int32Array>()
+                            .unwrap();
+                        self.rows += typed
+                            .write_batch(array.value_slice(0, array.len()), None, None)
+                            .unwrap() as i64;
+                    }
+                    //TODO add other types
```

Comment:
A couple of initial things to keep in mind

* Writes of both Nullable (OPTIONAL) and non-nullable (REQUIRED) fields
* You can optimize the special case where a nullable field's data has no nulls 
* A good amount of code is required to handle converting from the Arrow physical form of various logical types to the Parquet equivalent one, see https://github.com/apache/arrow/blob/master/cpp/src/parquet/column_writer.cc for details
* It would be worth thinking up front about how dictionary-encoded data is handled both on the Arrow write and Arrow read paths. In parquet-cpp we initially discarded Arrow DictionaryArrays on write, and through real world need I was forced to revisit this (quite painfully) to enable Arrow dictionaries to survive roundtrips to Parquet format, and also achieve better performance and memory use in both reads and writes. You can certainly do a dictionary-to-dense conversion like we did, but you may someday find yourselves doing the same painful refactor that I did to make dictionary write and read not only more efficient but also dictionary order preserving.